samples/nrf9160/modem_shell/esp_8266_nrf9160ns.overlay

```diff
@@ -0,0 +1,44 @@
+/* Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+ /*
+  * DTS overlay to be used when connecting nRF9160 DK with ESP8266-01 using
+  * the esp_at driver.
+  *
+  * https://en.wikipedia.org/wiki/ESP8266
+  *
+  * Wiring:
+  * - Slide (SW9) to enable 3V mode.
+  * - ESP8266 RX → DK P0.14 (TX)
+  * - ESP8266 TX → DK P0.15 (RX)
+  * - ESP8266 VCC → DK VDD
+  * - ESP8266 CH_PD → DK VDD
+  * - ESP8266 GND → DK GND.
+  *
+  * Compilation with SW overlay:
+  * 'west build -p -b nrf9160dk_nrf9160_ns -- -DDTC_OVERLAY_FILE=esp_8266_nrf9160ns.overlay -DOVERLAY_CONFIG="overlay-esp-wifi.conf"'
```

Comment:
I think it'd be reasonable to cover this configuration in `sample.yaml`, otherwise, it won't be tested by the CI.

Reply:
Done and build seemed to run that also.